lib/server/error/GrpcError.js

```diff
@@ -0,0 +1,51 @@
+class GrpcError extends Error {
+  /**
+   * @param {string} message
+   * @param {number} code
+   * @param {grpc.Metadata} [metadata]
+   */
+  constructor(code, message, metadata = undefined) {
+    super(message);
+
+    this.code = code;
+
+    if (metadata) {
+      this.metadata = metadata;
+    }
+  }
+
+  /**
+   * Get message
+   *
+   * @return {string}
+   */
+  getMessage() {
+    return this.message;
+  }
+
+  /**
+   * Get error code
+   *
+   * @return {number}
+   */
+  getCode() {
+    return this.code;
+  }
+
+  /**
+   * Get metadata
+   *
+   * @return {Object}
```

Comment:
It's `grpc.Metadata` in the constructor for some reason? Please correct me if I wrong. We operate with plain objects (it's easier) and convert to Metadata instance inside the client, right?